### PR TITLE
Disable left/right navigation when highlighting textarea.

### LINKED ISSUE #918: Disable left/right navigation when highlighting textarea
**Is your feature request related to a problem? Please describe.**
On https://www.neonlaw.com/bar-prep/flashcards when typing an answer, the `keydown` event listener for the `left` and `right` arrow keys should be removed.

### PR DESCRIPTION
**Pull request by nisarhassan12**
Fixes #918

### CODE REVIEW

File: packages/shared-ui/src/components/flashcardContainer.tsx [hunk 1]
```diff
@@ -22,35 +24,46 @@ export const FlashcardContainer = () => {
...
 
-  const keyDownListener = useCallback((event) => {
-    // keyCode 37 is the left arrow
-    if (event.keyCode === 37) {
-      if (selectedFlashcard === 0) {
-        return changeSelectedFlashcard(flashcards.length - 1);
-      }
-      if (selectedFlashcard > 0) {
-        return changeSelectedFlashcard(selectedFlashcard - 1);
+  const keyDownListener = useCallback(
```
Comment (nisarhassan12) [inline packages/shared-ui/src/components/flashcardContainer.tsx:30]: @shicholas I was not familiar with the `useCallback` hook i.e that it only runs when one of its dependencies change and not on every re-render. Due to this, I spent a lot of time debugging and figure out what is it that I'm doing wrong learning about the `useCallback` helped me understand that.

Also In this PR, I have fixed some other bugs in the flashcard component.

**Comment by nisarhassan12** [comment]
The build is failing with the following error. I tried investigating it but so far unable to find the cause

```
WebpackError: TypeError: Cannot create property 'context' on string 'Operating System not recognized.'
```

image

I have opened an issue for this https://github.com/gatsbyjs/gatsby/issues/27275

**Comment by shicholas** [comment]
> The build is failing with the following error. I tried investigating it but so far unable to find the cause
> 
> ```
> WebpackError: TypeError: Cannot create property 'context' on string 'Operating System not recognized.'
> ```
> 
> image
> 
> I have opened an issue for this [gatsbyjs/gatsby#27275](https://github.com/gatsbyjs/gatsby/issues/27275)

okay, np we can stop throwing that exception for now.

**Comment by nisarhassan12** [comment]
> okay, np we can stop throwing that exception for now.

@shicholas Where is the code for that exception? Or did you meant something else?

**Comment by shicholas** [comment]
It's this one you added I think https://github.com/NeonLaw/codebase/pull/899/files#diff-2d583ec9ffb4d6561a8125ea1918d9a0R20

**Comment by nisarhassan12** [comment]
Oh! Ok. My bad. I forgot about it. I'll close the issue in the Gatsby repo and will comment out the exception.

**Comment by shicholas** [comment]
> Oh! Ok. My bad. I forgot about it. I'll close the issue in the Gatsby repo and will comment out the exception.

no worries! programming is always easier on a team. I make mistakes like this all the time.

**Pull request merged**